generator/osm_source.cpp

```diff
@@ -524,6 +532,30 @@ class MainFeaturesEmitter : public EmitterBase
 };
 }  // anonymous namespace
 
+void SyncOfstream::Open(string const & fullPath)
+{
+  lock_guard<mutex> gard(m_mutex);
+  m_stream.open(fullPath, std::ofstream::out);
+}
+
+bool SyncOfstream::IsOpened()
+{
+  lock_guard<mutex> gard(m_mutex);
+  return m_stream.is_open() && !m_stream.fail();
+}
+
+void SyncOfstream::Write(uint32_t featureId, vector<osm::Id> const & osmIds)
+{
+  if (!IsOpened())
+    return;
+
+  lock_guard<mutex> gard(m_mutex);
+  m_stream << featureId << ", ";
```

Comment:
The third advantage of keeping the mapping in text files is the mapping from feature id to osm ids could be useful for other parts of our system. (@mgsergio has mentioned that it could be useful for him.)